src/main/java/seedu/address/logic/parser/ContactDetailsParser.java

```diff
@@ -41,7 +41,7 @@ public Command parseCommand(String userInput, Person person) throws ParseExcepti
         switch (commandWord) {
 
         case EditCommand.COMMAND_WORD:
-            return new EditCommandParser().parse(arguments);
+            return new EditCommandParser().parse(arguments).parsePerson(person);
```

Comment:
Could we not make EditCommandParser().parse() also take in person as a param and just pass it to EditCommand?

Reply:
possible yes. But since EditCommandParser implements Parser, still a need to define a parse(String) method. I don't think it is good practise to define a separate parse(String, Person) method while leaving a parse(String) method there so I created a parsePerson method to link with the existing parse method